react/Paywall/OnlyOfficePaywall.jsx

```diff
@@ -0,0 +1,108 @@
+import React from 'react'
+import ReactMarkdown from 'react-markdown'
+
+import { useClient } from 'cozy-client'
+import {
+  arePremiumLinksEnabled,
+  buildPremiumLink
+} from 'cozy-client/dist/models/instance'
+
+import useInstance, { useInstanceState } from '../helpers/useInstance'
+import withLocales from '../I18n/withLocales'
+import fr from './locales/fr.json'
+import en from './locales/en.json'
+import { IllustrationDialog } from '../CozyDialogs'
+import { Spinner } from 'cozy-ui/react'
+import Icon from '../Icon'
+import CozyUpgradeIcon from '../Icons/CozyUpgrade'
+import Button from '../Buttons'
+import Typography from '../Typography'
+
+const locales = {
+  en,
+  fr
+}
+
+const OnlyOfficePaywall = ({ onClose, isPublic, t }) => {
+  const client = useClient()
+  const instance = useInstance(client)
+
+  const hasPremiumEnabled = arePremiumLinksEnabled(instance)
+
+  if (instance.state === useInstanceState.Idle) return null
+
+  if (
+    instance.state === useInstanceState.Loading &&
+    instance.state !== useInstanceState.Loaded
+  ) {
+    return (
+      <IllustrationDialog
+        size="small"
+        actionsLayout="column"
+        open={true}
+        content={
+          <div className="u-h-5">
+            <Spinner size="xxlarge" noMargin middle />
+          </div>
+        }
+        onClose={onClose}
+      />
+    )
+  }
+
+  const link = buildPremiumLink(instance)
+  let type = 'default'
+  if (hasPremiumEnabled && isPublic) {
+    type = 'public'
+  } else if (hasPremiumEnabled && link) {
+    type = 'premiumLink'
+  }
+
+  const content =
+    type === 'default' ? (
+      <ReactMarkdown
+        source={t('onlyOfficePaywall.default.content', {
+          mail: instance?.context?.data?.attributes?.reply_to
+        })}
+        renderers={{
+          paragraph: props => <span className="u-mv-0">{props.children}</span>
+        }}
+      />
+    ) : (
+      t(`onlyOfficePaywall.${type}.content`)
+    )
+
+  const onAction = () => {
```

Comment:
I need an window.open because I open an external website. Also for the moment paywalls aren't allowed on AA or other mobile apps because we don't have in app payment

Reply:
> Also for the moment paywalls aren't allowed on AA or other mobile apps because we don't have in app payment

Yep but we expect that we have nothing to do to make them work. @Ldoppea with the recent evolution of the flagship app (webview url change interception and then call openApp), still we need to avoid `window.open`?

Reply:
Best practice is to use the cozy provided APIs like AppLinker & co so we can easily predict the behaviour now and in the future (i.e. if we decide to change the way we handle navigation between apps)

But on Flagship app we still catch `window.open()` and open it with an `InAppBrowser`. So in that case it should work as expected.

Reply:
Tu dis qu'on "still catch" , est-ce que ça risque de changer à l'avenir @Ldoppea ? Faut-il mettre un commentaire dans le code de penser à faire autrement un jour ? Est-ce qu'on ne peut pas tout de suite utiliser une méthode de AppLinker pour éviter de se poser ces questions ? Ou est-ce que window.open est pérenne ?

Reply:
> But on Flagship app we still catch window.open() and open it with an InAppBrowser. So in that case it should work as expected.

Here this is not want we want to open it in InAppBrowser since this is on the Cozy's url. `InAppBrowser` should only be used for the outside world.

> Best practice is to use the cozy provided APIs like AppLinker & co so we can easily predict the behaviour now and in the future (i.e. if we decide to change the way we handle navigation between apps)

Yes and no. We're still in the early day of the flagship app, but in the future, we want the third dev to be able to develop a cozy app without thinking about the flagship. The flagship should handle all the navigation by itself. But since we're still in the early day, we can still accept the use of cozy specific api ;)

Reply:
> Here this is not want we want to open it in InAppBrowser since this is on the Cozy's url. InAppBrowser should only be used for the outside world.

I reacted to the following assertion from @cballevre :
> I need an window.open because I open an external website

If the given URL is the cozy's one, then we redirect to `openApp`. If it's outside of the cozy, then we open it in an `InAppBrowser`.

Reply:
Oh right, @cballevre opens the cloudery not the settings page of the cozy (yet). 

Thanks it's clear now, and we can keep the use of window.open ;)

Reply:
For more details the implementation is here: https://github.com/cozy/cozy-flagship-app/blob/master/src/components/webviews/ReloadInterceptorWebView.js#L110-L152

```js
const interceptOpenWindow = ({
  currentUrl,
  destinationUrl,
  subdomainType,
  navigation,
  webViewForwardRef
}) => {
  //...

  if (isSameApp) {
    navigateTo(webViewForwardRef, destinationUrl)
    return
  }

  //...

  if (newSlug) {
    navigateToApp({
      navigation,
      href: destinationUrl,
      slug: newSlug
    })
    return
  }

  //...

  if (isRedirectOutside) {
    openUrlInAppBrowser(destinationUrl)
    return
  }
}
```